src/optim/FishLeg/fishleg_likelihood.py

```diff
@@ -61,6 +66,12 @@ def draw(self, preds, **kwargs):
         """
         raise NotImplementedError
 
+    def get_parameters(self) -> List:
```

Comment:
Is this an abstract method? If we are requiring the list of learnable parameters to be defined (even if there may not be any) then we should probably decorate this and follow the `raise NotImplementedError`.

Basically, if someone defines a new likelihood and forgets to do this then it might be a hidden error if we don't through a `NotImplementedError`.

If it is non-essential then this is OK - I would just rewrite it as returning None.

Reply:
Because in the fishleg main optimiser, I will check whether the length of this function is > 0. So I return [] as default in here. If someone defines a new likelihood with parameters, they should return the parameters as a list in here.

Reply:
OK cool, so in that case, I think we should have it as a `raise NotImplementedError` in the FishLiklihood abstract class. Then define it explicitly for each likelihood below.

Since otherwise this may be missed in defining new likelihood classes.